src/components/PopupConfirmation.js

```diff
@@ -0,0 +1,38 @@
+import Popup from "./Popup";
+
+export default class PopupConfirmation extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._deleteConfirm = this._popupElement.querySelector(
+      "#confirm-delete-button"
+    );
+    /*this._handleDeleteConfirm = null; */
+  }
+
+  setSubmit(confirmAction) {
+    this._handleDeleteConfirm = confirmAction;
+  }
+
+  /*setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      if (this._handleDeleteConfirm) {
+        this.setSubmit();
+      } else {
+        console.error();
+      }
+      this.close();
+    });
+  }*/
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleDeleteConfirm();
+      this.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
I fixed the issue I was having for submitting cards when I added a then block in my PopupWithForm, but im still getting an error when I my delete popup comes up after adding a then block

Reply:
nevermind no longer seeing error